plugins/elanfp/fu-elanfp-usb-device.c

```diff
@@ -0,0 +1,564 @@
+			    "received length (%d) is not match with the request (%d)",
+			    (gint32)actual,
+			    (gint32)len);
+
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+gboolean
+runIapProcess(FuElanfpUsbDevice *self, GBytes *fw, GError **error)
+{
+	const guint8 *pbinary = NULL;
+	guint8 databuf[61] = {0};
+	guint8 recvbuf[17] = {0};
+	S2FFILE s2ffile;
+	PAYLOAD_HEADER *ppayload_header = NULL;
+	guint32 payloadheader_length = 5;
+	guint32 pkg_index = 0;
+	guint32 payload_offset = 0;
+	gsize binary_size;
+	gboolean result = TRUE;
+	GUsbDevice *usb_device = fu_usb_device_get_dev(FU_USB_DEVICE(self));
+
+	if (fw == NULL)
+		return FALSE;
+
+	pbinary = g_bytes_get_data(fw, &binary_size);
+
+	if (pbinary == NULL) {
+		// g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_INTERNAL, "device write fw - binary
+		// buffer is null !!\n");
+		return FALSE;
+	}
+
+	if (!binaryVerify(pbinary, binary_size, &s2ffile, error)) {
+		// g_set_error(error, FWUPD_ERROR, FWUPD_ERROR_INVALID_FILE, "device write fw -
+		// binary verify fail !!\n");
+		return FALSE;
+	}
+
+	// Send Offer
+	for (int i = 0; i < 2; i++) {
+		memset(databuf, 0x00, sizeof(databuf));
+		memset(recvbuf, 0x00, sizeof(recvbuf));
+		databuf[0] = REPORT_ID_OFFER_COMMAND;
+		memcpy(databuf + 1, s2ffile.pOffer[i], s2ffile.OfferLength[i]);
+
+		if (!iapSendCommand(usb_device,
+				    REQTYPE_COMMAND,
+				    REPORT_ID_OFFER_COMMAND,
+				    databuf,
+				    s2ffile.OfferLength[i] + 1,
+				    error)) {
+			result = FALSE;
+			goto Exit;
+		}
+
+		if (!iapRecvStatus(usb_device, recvbuf, s2ffile.OfferLength[i] + 1, error)) {
+			result = FALSE;
+			goto Exit;
+		}
+
+		if (recvbuf[13] == FIRMWARE_UPDATE_OFFER_ACCEPT) {
+			// Payload
+
+			g_debug("*******************************************************\n");
+			g_debug("run iap - Offer %s Accepted\n", s2ffile.Tag[i]);
+			g_debug("*******************************************************\n");
+
+			pkg_index = 1;
+			payload_offset = 0;
+			while (payload_offset < s2ffile.PayloadLength[i]) {
+				memset(databuf, 0x00, sizeof(databuf));
+				memset(recvbuf, 0x00, sizeof(recvbuf));
+
+				ppayload_header =
+				    (PAYLOAD_HEADER *)(s2ffile.pPayload[i] + payload_offset);
+
+				databuf[0] = REPORT_ID_PAYLOAD_COMMAND;
+				databuf[1] =
+				    (pkg_index == 1)
+					? FIRMWARE_UPDATE_FLAG_FIRST_BLOCK
+					: ((payload_offset + payloadheader_length +
+					    ppayload_header->Length) >= s2ffile.PayloadLength[i]
+					       ? FIRMWARE_UPDATE_FLAG_LAST_BLOCK
+					       : 0x00);
+				databuf[2] = ppayload_header->Length;
+				*(guint16 *)(databuf + 3) = pkg_index;
+				*(guint32 *)(databuf + 5) = ppayload_header->Address;
+
+				memcpy(databuf + 9,
+				       s2ffile.pPayload[i] + payload_offset + payloadheader_length,
+				       databuf[2]);
+
+				if (!iapSendCommand(usb_device,
+						    REQTYPE_COMMAND,
+						    REPORT_ID_PAYLOAD_COMMAND,
+						    databuf,
+						    sizeof(databuf),
+						    error)) {
+					result = FALSE;
+					goto Exit;
```

Comment:
not `goto Exit`, but return FALSE please

Reply:
Fixed in commit 5af9c23594ac261d85fbf53fca02de08a5ef0150.